server/middleware/isAuth.js

```diff
@@ -1,36 +1,52 @@
 require('dotenv').config();
 
-const responseMessage = require('./../helpers/responseMessage');
+const responseMessage = require('../helpers/responseMessage');
 
 const { verify } = require('jsonwebtoken');
 
+const { getUserById } = require('../database/query/user');
+
 module.exports = (req, res, next) => {
   const token = !req.cookies ? null : req.cookies.AuthToken;
   if (!token) {
     return res
-      .status(200)
+      .status(403)
       .json(
         responseMessage.UnauthorizedMessage(
           null,
-          'please login to continue... '
-        )
+          'please login to continue... ',
+        ),
       );
   }
 
   verify(token, process.env.acces_Token_secret, (err, payload) => {
     if (err) {
       return res
-        .status(200)
+        .status(403)
         .clearCookie('AuthToken')
         .json(
           responseMessage.UnauthorizedMessage(
             null,
-            'please login to continue... '
-          )
+            'please login to continue... ',
+          ),
         );
     }
 
-    req.user = payload;
-    return next();
+    getUserById(payload.id)
```

Comment:
A small code style thing - it will make it easier as the code base gets bigger if you make careful decisions about what each function returns

For example - to use this function, you have to know that the database driver that is being used will return an object with a `rows` array property.  It can be a source of bugs and give you some extra work to do if you ever want to switch data sources

It's good practice to delegate everything related to database code to within the db layer (in your code, everything inside `/server/database`. I think a nice design is for getUserById to return a Promise that resolves with either:

1. a single User object
2. null

Then it makes identifying the error cases a bit easier - what should happen if the user wasn't found here?

Reply:
in gsg we learned this way to handle server error outside database layer  ,
where the database execute the query and return the promise , to handle result outside database
 If there is a way to handle error ,  without using callback , please share with us 

in this case if the user not found i clean cookie and the user must re-login to continue "security reason"